thirdparty/build-thirdparty.sh

```diff
@@ -1205,6 +1205,36 @@ build_libdeflate() {
     ${BUILD_SYSTEM} install
 }
 
+#clucene
+build_clucene() {
+
+    check_if_source_exist "${CLUCENE_SOURCE}"
+    cd "$TP_SOURCE_DIR/${CLUCENE_SOURCE}"
+
+    mkdir -p "${BUILD_DIR}"
+    cd "${BUILD_DIR}"
+    rm -rf CMakeCache.txt CMakeFiles/
+
+    ${CMAKE_CMD} -G "${CMAKE_GENERATOR}" \
+        -DCMAKE_INSTALL_PREFIX="$TP_INSTALL_DIR" \
+        -DCMAKE_INSTALL_LIBDIR=lib64 \
+        -DBUILD_STATIC_LIBRARIES=ON \
+        -DBUILD_SHARED_LIBRARIES=OFF \
+        -DBOOST_ROOT="$TP_INSTALL_DIR" \
+        -DZLIB_ROOT="$TP_INSTALL_DIR" \
+        -DCMAKE_CXX_FLAGS="-g -fno-omit-frame-pointer -Wno-narrowing" \
+        -DUSE_STAT64=0 \
+        -DCMAKE_BUILD_TYPE=Release \
+        -DBUILD_CONTRIBS_LIB=ON ..
+    ${BUILD_SYSTEM} -j "${PARALLEL}"
+    ${BUILD_SYSTEM} install
```

Comment:
this install will include additional boost headers, a little worried about this, will it cause boost header conflict?

Reply:
I did not meet any conflict when I compiled and tested, but your worry makes sense, I will keep an eye on this issue and try to address it in the subsequent refactoring.